data_fetcher.py

```
@@ -0,0 +1,45 @@
+import bs4
+import requests
+import lxml
+from bs4 import BeautifulSoup
+from typing import List
+ 
+ 
+class JeuneAfriqueRSSDataSource:
+    
+    def __init__(self, rss_url: str) -> None:
+        self.url = rss_url
+        self.r = None
+        self.status_code = None
+        self.soup = None
+        self.articles = None
+        self.articles_dicts = None
+        self.urls: List[str] = []
+        self.titles: List[str] = []
+        self.descriptions: List[str] = []
+        self.pub_dates: List[str] = []
+        
+    def get_data(self) -> None:
+        try:
+            self.r = requests.get(self.url)
+            self.status_code = self.r.status_code
+        except Exception as e:
+            print("Error fetching the URL: ", self.url)
+            print(e)
+        try:
+            self.soup = BeautifulSoup(self.r.text, "lxml")
+        except Exception as e:
```

same here. Failing explicitly is better than swallowing an error and continuing